lang/src/org/partiql/lang/eval/ExprValueFactory.kt

```diff
@@ -292,6 +299,29 @@ private class DecimalExprValue(val ion: IonSystem, val value: BigDecimal): Scala
     override fun ionValueFun(): IonValue = ion.newDecimal(value)
 }
 
+/**
+ * [ExprValue] to represent DATE in PartiQL.
+ * [LocalDate] represents date without time and time zone.
+ */
+private class DateExprValue(val ion: IonSystem, val value: LocalDate): ScalarExprValue() {
+
```

Comment:
Inside of an `init { }` might be a good place to validate that the year is between 1 and 9999.  This should be like a last-resort check tho, not to take place of other validations I've mentioned elsewhere.

Reply:
I *would not* put that here unless you want to make this an invariant in the data model.  I don't feel strongly either way, and maybe the extra defensiveness is good, but either way let's consolidate that logic.

Reply:
I think it's okay to safeguard and have this check in the `init {}` block. Added. Thanks.